include/nbl/builtin/hlsl/fft/common.hlsl

```diff
@@ -1,34 +1,46 @@
 #ifndef _NBL_BUILTIN_HLSL_FFT_COMMON_INCLUDED_
 #define _NBL_BUILTIN_HLSL_FFT_COMMON_INCLUDED_
 
-#include "nbl/builtin/hlsl/complex.hlsl"
+#include "nbl/builtin/hlsl/math/complex.hlsl"
 #include "nbl/builtin/hlsl/cpp_compat.hlsl"
-#include "nbl/builtin/hlsl/fft/common/twiddles.hlsl"
+#include "nbl/builtin/hlsl/math/numbers.hlsl"
 
 namespace nbl 
 {
 namespace hlsl
 {
-namespace fft::common {
+namespace fft {
+
+    // Computes the kth element in the group of N roots of unity
+    template<typename Scalar, bool inverse>
+    complex_t<Scalar> twiddle(uint32_t k, uint32_t N){
+        complex_t<Scalar> retVal;
+        retVal.real( cos(2.f * math::Numbers<Scalar>::pi * float(k) / N) );
+        if (! inverse)
+            retVal.imag( sin(2.f * math::Numbers<Scalar>::pi * float(k) / N) );
+        else
+            retVal.imag( sin(-2.f * math::Numbers<Scalar>::pi * float(k) / N) );
```

Comment:
1. don't add extra indent because of namespace
2. use the `{` on newline convention
3. hoist the computation of `2.f*math::Numbers<Scalar>::pi/Scalar(N)` into a variable on line 17
4. its not `float(k)` it should be `Scalar(k)` - style nitpick, nothing really will happen